cardano-node/src/Cardano/Node/Handlers/Shutdown.hs

```diff
@@ -136,20 +121,30 @@ withShutdownHandling
   -> (ShutdownFDs -> IO ())
   -> IO ()
 withShutdownHandling cli trace action = do
-  sfds <- decideShutdownFds cli
-  withShutdownHandler sfds trace (action sfds)
- where
-   decideShutdownFds :: NodeCLI -> IO ShutdownFDs
-   decideShutdownFds NodeCLI{shutdownIPC = Just fd} =
-     pure $ ExternalShutdown (ShutdownListener fd)
-   decideShutdownFds NodeCLI{shutdownOnSlotSynced = MaxSlotNo{}} =
-     mkInternalShutdown
-   decideShutdownFds _ = pure NoShutdownFDs
-
-   mkInternalShutdown :: IO ShutdownFDs
-   mkInternalShutdown = do
-     (r, w) <- IO.createPipeFd
-     pure $ InternalShutdown (ShutdownListener $ Fd r) (ShutdownDoorbell $ Fd w)
+  sfds <- case cli of
+    NodeCLI{shutdownIPC = Just fd} -> pure $ ExternalShutdown (ShutdownListener fd)
+    NodeCLI{shutdownOnSlotSynced = MaxSlotNo{}} -> do
+      (r, w) <- IO.createPipeFd
+      pure $ InternalShutdown (ShutdownListener $ Fd r) (ShutdownDoorbell $ Fd w)
```

Comment:
After this refactoring, it became obvious that the `ShutdownDoorbell` constructed here is ignored below at `_w` and so is unused.  Did we forget to use it or is the `ShutdownDoorbell` redundant?

Reply:
I believe @deepfire created this module

Reply:
@newhoggy, not ignored -- the `ShutdownFDs` are passed to `action`, where the write side of the doorbell can be used (i.e. closed).